pkg/driver/controller.go

```diff
@@ -126,17 +159,39 @@ func CreateZFSVolume(req *csi.CreateVolumeRequest) (string, error) {
 
 	vtype := zfs.GetVolumeType(fstype)
 
+	capacity := strconv.FormatInt(int64(size), 10)
+
+	if vol, err := zfs.GetZFSVolume(volName); err == nil {
+		if vol.DeletionTimestamp != nil {
+			if _, ok := parameters["wait"]; ok {
+				if err := waitForVolDestroy(volName); err != nil {
+					return "", err
+				}
+			}
+		} else {
+			if vol.Spec.Capacity != capacity {
+				return "", status.Errorf(codes.AlreadyExists,
+					"volume %s already present", volName)
+			}
+			return vol.Spec.OwnerNodeID, nil
+		}
+	}
+
```

Comment:
Should we add a return for errors other than NotFound.

Reply:
this is Create request. We don't need to return any other error as we are creating the volume. The above check is added for sanity to check if volume is already present with the same capacity, then return successful response.